### DISCUSSION TITLE
Unique style classes in defs

### DISCUSSION THREAD

**Discussion by CarlosAmaral** [gilbarbara/react-inlinesvg#196] [Q&A] [answered]
Hi,

Thank you for this amazing library!
Is it possible to apply the same concept of unique ids in class name styles in `defs` for example. Below follows an example:

`
<defs><style>.cls-1{fill:#030303;}.cls-2{fill:#fff;}.cls-3{fill:#ee262f;}.cls-4{fill:#f3bbd2;}.cls-5{fill:#e7292f;}.cls-6{fill:#e52931;}.cls-7{fill:#ad3039;}.cls-8{fill:#b13034;}.cls-9{fill:#9d303f;}.cls-10{fill:#bc2c39;}.cls-11{fill:#842729;}.cls-12{fill:#060606;}.cls-13{fill:#010101;}.cls-14{fill:#f9bd1d;}</style></defs>
`

The idea would be to have unique classes in this case.

**Comment by gilbarbara** [comment, marked as answer]
Hey @CarlosAmaral 

Since the content of style tags isn't very structured, it would be hard to get it right.
You can try the `preProcessor` handler to update that by yourself.

**Reply by CarlosAmaral** [reply]
Hi @gilbarbara,

I've used the preProcessor handler indeed with the svgo-browser's optimize function by passing some arguments. It worked great. Most likely in the future all the svg preprocessing will be done on the backend so svgo doesnt need to pre-process them every single time they're loaded. Thank you for the answer
